x/meicdp/keeper/keeper.go

```diff
@@ -0,0 +1,123 @@
+package keeper
+
+import (
+	"encoding/binary"
+	"fmt"
+
+	"github.com/bandprotocol/bandchain/chain/x/oracle"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/tharamalai/meichain/x/meicdp/types"
+)
+
+type Keeper struct {
+	storeKey      sdk.StoreKey
+	cdc           *codec.Codec
+	BankKeeper    types.BankKeeper
+	ChannelKeeper types.ChannelKeeper
+}
+
+// NewKeeper creates a new band consumer Keeper instance.
+func NewKeeper(cdc *codec.Codec, key sdk.StoreKey, channelKeeper types.ChannelKeeper, bankKeeper types.BankKeeper) Keeper {
+	return Keeper{
+		storeKey:      key,
+		cdc:           cdc,
+		ChannelKeeper: channelKeeper,
+		BankKeeper:    bankKeeper,
+	}
+}
+
+func (k Keeper) SetResult(ctx sdk.Context, requestID oracle.RequestID, result []byte) {
+	store := ctx.KVStore(k.storeKey)
+	store.Set(types.ResultStoreKey(requestID), result)
+}
+
+func (k Keeper) GetResult(ctx sdk.Context, requestID oracle.RequestID) ([]byte, error) {
+	if !k.HasResult(ctx, requestID) {
+		return nil, sdkerrors.Wrapf(types.ErrItemNotFound,
+			"GetResult: Result for request ID %d is not available.", requestID,
+		)
+	}
+	store := ctx.KVStore(k.storeKey)
+	return store.Get(types.ResultStoreKey(requestID)), nil
+}
+
+func (k Keeper) HasResult(ctx sdk.Context, requestID oracle.RequestID) bool {
+	store := ctx.KVStore(k.storeKey)
+	return store.Has(types.ResultStoreKey(requestID))
+}
+
+//SetCDP - set CDP of user account to the store
+func (k Keeper) SetCDP(ctx sdk.Context, cdp types.CDP) {
+	store := ctx.KVStore(k.storeKey)
+	c := k.cdc.MustMarshalBinaryBare(cdp)
+	store.Set(types.CDPStoreKey(cdp.Owner), c)
+}
+
+//GetCDP - get CDP of user account from the store
+func (k Keeper) GetCDP(ctx sdk.Context, account sdk.AccAddress) (types.CDP, error) {
```

Comment:
GetCDP should return the default CDP if it does not exist. Make the design simpler.

Reply:
fixed